### ISSUE TITLE
Inconsist diagnostics on extension GetAwaiter

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn#39155]
After invesgating #25861, I've found that using `await` when missing `GetAwaiter()` *extension* method can results in 3 different error codes:
- **CS1061** No such instance or extension member
```csharp
    public class Foo
    {
        async void M(Foo foo)
        {
            await foo;
        }
    }
```
- **CS4036** Specialised  for WinRT async interfaces
```csharp
    public class Foo
    {
        async void M(Windows.Foundation.IAsyncAction ia)
        {
            await ia;
        }
    }
```
- **CS1929** Extension method overload resolution fail
```csharp
using System;
using System.Runtime.CompilerServices;

namespace A
{
    public class Foo
    {
        async void M(Foo foo)
        {
            await foo;
        }
    }

    public static class BarExtension
    {
        public static Extension.FooAwaiter GetAwaiter(this string s) => default;
    }
}

namespace A.Extension
{
    public static class FooExtensions
    {
        public static FooAwaiter GetAwaiter(this Foo foo) => default;
    }

    public struct FooAwaiter : INotifyCompletion
    {
        public bool IsCompleted { get; }
        public void OnCompleted(Action continuation) {}
        public void GetResult() {}
    }
}

```

Currently, add imports service can only recognize the first case **CS1061**.
Linq syntax provides a specialised **CS1929** and the general **CS1936**.
However, missing method for deconstruction provides one consist error code **CS8129**:
```csharp
using System;
using System.Runtime.CompilerServices;

namespace A
{
    public class Foo
    {
        async void M(Foo foo)
        {
            var (a, b) = foo;
        }
    }

    public static class BarExtension
    {
        // Reports CS8129 no matter existence of this method
        public static void Deconstruct(this string foo, out int a, out int b)
        {
            a = 1;
            b = 2;
        }
    }
}

namespace A.Extension
{
    public static class FooExtensions
    {
        public static void Deconstruct(this Foo foo, out int a, out int b)
        {
            a = 1;
            b = 2;
        }
    }
}
```
This makes code fix much easier to deal with it.
In all the three syntaxes, diagnostics span differs on existence of extension on other type.

Can we unify the error codes and diagnostics span for such scenarios?

**Comment by CyrusNajmabadi** [comment]
> Can we unify the error codes and diagnostics span for such scenarios?

Probably not.  That could break existing consumers that depend on those values/locations.

**Comment by huoyaoyuan** [comment]
For Visual Basic, there's a consist error `BC36930` for `await`. But the code fix handles `BC36610`. The error code should have changed some time ago.

**Comment by gafter** [comment]
@huoyaoyuan We don't have plans to change the diagnostics.
